### PR TITLE
cloud-config user-data:  ntp to provide ntp_client: ntpsec support

### PR DESCRIPTION
**Pull request by blackboxsw**
Due to the ntp being a transitional package in may install ntpd or ntpsec on different distributions, it may be preferred 
to specify ntpsec as the preferred ntp_client in general to strictly define the client that an admin wishes to install/enable.  Provide a mechanism to directly prefer a specific  `ntp_client: ntpsec` instead of the ntp transitional package.

Also, in Ubuntu, daily integration tests against development Ubuntu series will occasionally hit issues with APT dependency resolution when trying to provide user-data declaring a preference for ntp transitional package when systemd-timesyncd packages in Ubuntu cloud images are out of date with latest published packages in the <release>-updates pocket. APT is unable to resolve a stale systemd-timesyncd package as it `Provides: time-daemon` which conflicts with `ntpsec` which also `Provides: time-daemon`.

## Proposed Commit Message
 see individual commits
* pytestify tests/unittests/config/test_cc_ntp.py
* Add `ntp_client: ntpsec` support to #cloud-config

## Additional Context

Intermittent yet repeated ntp error on ubuntu devel releases
https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/48/
```
tests.integration_tests.modules.test_ntp_servers.TestNtpServers.test_ntp_installed (from pytest)
Failing for the past 3 builds (Since Unstable
[#46](https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/46/) )
[Took 20 sec.](https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/48/testReport/junit/tests.integration_tests.modules.test_ntp_servers/TestNtpServers/test_ntp_installed/history)
Error Message

AssertionError: assert False
 +  where False = ''.ok
 +    where '' = execute('ntpd --version')
 +      where execute = <tests.integration_tests.instances.IntegrationInstance object at 0x7fb95540a0b0>.execute
```

## Test Steps
```
CLOUD_INIT_OS_IMAGE=noble CLOUD_INIT_CLOUD_INIT_SOURCE=IN_PLACE tox -e integration-tests -- tests/integration_tests/modules/test_ntp_servers.py::TestNtpServers
CLOUD_INIT_OS_IMAGE=oracular CLOUD_INIT_CLOUD_INIT_SOURCE=IN_PLACE tox -e integration-tests -- tests/integration_tests/modules/test_ntp_servers.py::TestNtpServers
```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: templates/ntpsec.conf.alpine.tmpl
Comment (dermotbradley) [inline templates/ntpsec.conf.alpine.tmpl:1]: Having this file doesn't make sense as Busybox's ntpd is limited and does not support NTPSEC.

ntpsec and openntpd are packages for Alpine but cloud-init doesn't currently support them (or ntpd-rs either, which I might consider adding support for in the future).

File: tests/integration_tests/modules/test_ntp_servers.py [hunk 1]
```diff
@@ -46,13 +46,17 @@ def test_dist_config_file_is_empty(
         (This test is skipped on all currently supported Ubuntu releases, so
         may not actually be needed any longer.)
         """
-        if class_client.execute("test -e /etc/ntp.conf.dist").failed:
-            pytest.skip("/etc/ntp.conf.dist does not exist")
-        dist_file = class_client.read_from_file("/etc/ntp.conf.dist")
+        if class_client.execute(
+            "test -e /etc/ntpsec/ntp.conf.dpkg-dist"
```
Comment (aciba90) [inline tests/integration_tests/modules/test_ntp_servers.py:50]: Shouldn't we operate over `/etc/ntpsec/ntp.conf.dist`?

```suggestion
            "test -e /etc/ntpsec/ntp.conf.dist"
```

`/etc/ntpsec/ntp.conf.dist` is the file cloud-init renames if `/etc/ntpsec/ntp.conf` is present and `/etc/ntpsec/ntp.conf.dpkg-dist` is the one `apt` / `dpkg` creates on installation / upgrade paths.

File: tests/integration_tests/modules/test_ntp_servers.py [hunk 2]
```diff
@@ -46,13 +46,17 @@ def test_dist_config_file_is_empty(
...
+        ).failed:
+            pytest.skip("/etc/ntpsec/ntp.conf.dpkg-dist does not exist")
+        dist_file = class_client.read_from_file(
+            "/etc/ntpsec/ntp.conf.dpkg-dist"
+        )
         assert 0 == len(dist_file.strip().splitlines())
```
Comment (aciba90) [inline tests/integration_tests/modules/test_ntp_servers.py:56]: This assertion fails, the file has content. Is this due to my previous comment?

It feels like somewhere while the test instance is being customized or the last boot, `ntpsec` gets reinstalled with `/etc/ntpsec/ntp.conf` present and modified and then `etc/ntpsec/ntp.conf.dpkg-dist` gets created as a backup by dpkg.

Is this a testing error or what is the intention with this test?

**Review by aciba90 — COMMENTED** [review]
General approach looks good, thanks @blackboxsw.

I left an inline question about an integration tests. In addition to that, could you please resolve the merge conflict?

**Pull request closed without merging**